release/ray_release/byod/byod.Dockerfile

```diff
@@ -24,3 +24,5 @@ EOF
 
 COPY "$PIP_REQUIREMENTS" .
 RUN "$HOME"/anaconda3/bin/pip install --no-cache-dir -r "${PIP_REQUIREMENTS}"
+
+ENV LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libjemalloc.so:$LD_PRELOAD
```

Comment:
is this the final form?

should we just set an `RAY_LD_PRELOAD`, and set this env var when ray is launching gcs or raylet? rather than setting this as global env?

Reply:
I think using `LD_PRELOAD` is fine given that the container is only for running ray workloads.

I'm fine either way. Defer to @can-anyscale on final decision.

Reply:
if we want this as a global feature for all release tests i think this is fine; this layer is specific to ray release tests

Reply:
Eventually we want to only set `LD_PRELOAD` for `ray start` command so that ONLY ray processes will use jemalloc. But I'm not sure how to do it here. I think this is good enough as a temporary thing given we will revert it back later and rely on other places to set `LD_PRELOAD`.

Reply:
this will enable jemalloc on all release tests but not in production?

so we will be testing something that is different from production?